src/Auth/OAuth.php

```diff
@@ -78,14 +85,33 @@ public function __construct(string $appId, string $appSecret, string $callback,
      */
     abstract public function getUserName(string $accessToken):string;
 
+    /**
+     * @param $scope
+     * 
+     * @return array
+     */
+    protected function addScope(string $scope){
+        // Add a scope to the scopes array if it isn't already present 
+        if (!in_array($scope, $this->userScopes)){
+            $this->userScopes[] = $scope; 
+        }
```

Comment:
We can use here:
```
return $this;
```

to allow chaining.

Reply:
Also need to change docblock return value

Reply:
could you please elaborate this? I dont think I understood

Reply:
when adding this statement:
```
return $this;
```

we can use this function this way (chaining):
```
$oauth
    ->addScope('x')
    ->addScope('x')
;
```

its very common to do this in PHP set and add methods

Reply:
Noted and done.